test/core/crawlers/basic_crawler.test.ts

```diff
@@ -196,32 +202,47 @@ describe('BasicCrawler', () => {
         ]);
     });
 
-    test('enqueueLinks should respect maxCrawlDepth', async () => {
-        const processedUrls: string[] = [];
+    describe('enqueueLinks should respect maxCrawlDepth', async () => {
+        const testMaxCrawlDepth = async (enqueueLinksOptions?: EnqueueLinksOptions) => {
+            const processedUrls: string[] = [];
 
-        const requestHandler: RequestHandler = async ({ request, enqueueLinks }) => {
-            processedUrls.push(request.url);
-            const url = new URL(request.url);
-            url.pathname = `${url.pathname}deep/`;
+            const requestHandler: RequestHandler = async ({ request, enqueueLinks }) => {
+                processedUrls.push(request.url);
+                const url = new URL(request.url);
+                url.pathname = `${url.pathname}deep/`;
 
-            await enqueueLinks({
-                urls: [url.toString()],
+                await enqueueLinks({
+                    urls: [url.toString()],
+                    ...enqueueLinksOptions,
+                });
+            };
+
+            const crawler = new BasicCrawler({
+                maxCrawlDepth: 2,
+                maxRequestsPerCrawl: 10, // safeguard against infinite loops
+                requestHandler,
             });
+
+            await crawler.run(['https://example.com/']);
+
+            expect(processedUrls).toEqual([
```

Comment:
A personal preference/nit - I like to keep the `expect` calls in the top-level on the test (not in the nested functions). While reading the test, you immediately see all the tested assertions. The `testMaxCrawlDepth` could return the `processedUrls` function (and we could assert on the returned value).

Note that this is optional, the current proposal is IMO still readable just fine :+1: